pkg/operator/discover/discover.go

```diff
@@ -241,6 +239,13 @@ func (d *Discover) createDiscoverDaemonSet(ctx context.Context, namespace, disco
 
 }
 
+func getLabels() map[string]string {
+	labels := make(map[string]string)
+	k8sutil.AddRecommendedLabels(labels, "rook-discover", "rook-ceph-operator", "rook-discover", os.Getenv("POD_NAME"))
```

Comment:
os.Getenv("POD_NAME") is only going to give the name of the operator pod. After some discussion with Travis, I think as long as this is unique per namespace, we are okay, and we can just set this to also be `rook-discover` with no ID or hash.

```suggestion
	k8sutil.AddRecommendedLabels(labels, "rook-discover", "rook-ceph-operator", "rook-discover", os.Getenv("POD_NAME"))
```